### DISCUSSION TITLE
Swagger string Field types with regex pattern not displaying properly

### DISCUSSION THREAD

**Discussion by dfrtz** [fastapi/fastapi#9521] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

from pydantic import BaseModel
from pydantic import Field


app = FastAPI(
    title='Repro',
    description='Repro weird type',
    version='0',
)

class TestRequest(BaseModel):
    broken_type: str = Field(
        title='Broken Type',
        description='Type looks like an odd hash instead of string',
        max_length=128,
        regex=r'[0-9a-zA-Z]',
        default=None
    )
    working_type: str = Field(
        title='Working Type',
        description='Type looks like normal string in Swagger',
        max_length=128,
        default=None,
    )


@app.post('/test')
def test_endpoint(
        request: TestRequest,
) -> dict:
    return {}
```

### Description

**Summary**
Starting in FastAPI 0.75.2, there appears to be a bug with string Field types showing random text in the type information.

**Details**
This may ultimately not be a FastAPI issue based on my investigation, however I am going to start by raising the question here in case it might have something to do with the way the OpenAPI specs are being generated. Also, it could come in handy for anyone else using the Swagger docs in FastAPI to know there is a workaround.

When the included code example is run, it ends up showing very odd (and random) results on any Fields in Models that use regex. This worked until 0.75.2 based on me binary searching the release/commit history, and after that point it starts to show scrambled results like following, with it changing each time:
```
{
  "broken_type": "f",
  "working_type": "string"
}

{
  "broken_type": "j",
  "working_type": "string"
}

{
  "broken_type": "W",
  "working_type": "string"
}
```

Where I would instead expect the behavior to continue to match 0.75.1:
```
{
  "broken_type": "string",
  "working_type": "string"
}
```

The following workaround appears to resolve the issue for anyone using FastAPI 0.75.2+, as it reverts the change from commit acf8a91c:
```
@app.get("/docs", include_in_schema=False)
async def get_swagger_docs():
    """Override Swagger /docs generation to force specific version."""
    # Force v3, but match the standard behavior of the automatically added /docs endpoint.
    return get_swagger_ui_html(
        openapi_url=app.openapi_url,
        title=f'{app.title} - Swagger UI',
        oauth2_redirect_url=app.swagger_ui_oauth2_redirect_url,
        swagger_js_url="https://cdn.jsdelivr.net/npm/swagger-ui-dist@3/swagger-ui-bundle.js",
        swagger_css_url="https://cdn.jsdelivr.net/npm/swagger-ui-dist@3/swagger-ui.css",
    )
```

So yes, now one might be saying: This is a Swagger issues since commit acf8a91c states "Upgrade Swagger UI - swagger-ui-dist@4", however I am just mostly looking for confirmation that we do not believe it has anything to do with the way FastAPI is generating the docs, and also attempting to bring visibility to the issue in case FastAPI would like to revert the change due what appears to be a regression in v4 if it is not FastAPI related.

### Operating System

Linux, macOS

### FastAPI Version

0.75.2+

### Python Version

3.10.2, 3.10.8

**Comment by YuriiMotov** [comment, marked as answer]
```
{
  "broken_type": "f",
  "working_type": "string"
}
```
The above JSON is not a schema, but just example values automatically generated based on parameter type.
So, "f" matches the regex pattern you defined (`r'[0-9a-zA-Z]'`)
